### ISSUE TITLE
Add enableKeepAlive property to ConnectionOptions interface 

### ISSUE THREAD

**Issue by luisasalas** [sidorares/node-mysql2#2286]
Noticed `enableKeepAlive` is missing on `Connection.d.ts`

**Comment by wellwelwel** [comment]
Comparing the [`ConnectionOptions`](https://github.com/sidorares/node-mysql2/blob/8831e092024f8d26fe9272adec8e1a5f115735aa/typings/mysql/lib/Connection.d.ts#L81-L300) and [`validOptions`](https://github.com/sidorares/node-mysql2/blob/8831e092024f8d26fe9272adec8e1a5f115735aa/lib/connection_config.js#L16-L69), both `enableKeepAlive` and `keepAliveInitialDelay` options are missing. Instead, they are in `PoolOptions`:
https://github.com/sidorares/node-mysql2/blob/8831e092024f8d26fe9272adec8e1a5f115735aa/typings/mysql/lib/Pool.d.ts#L41-L49

---

@sidorares, can you confirm if these options work for both `createConnection` and `createPool | createPoolCluster` or only when using *pools*?

**Comment by luisasalas** [comment]
@wellwelwel I assumed they should be since they are define in ConnectionConfig?

https://github.com/sidorares/node-mysql2/blob/8831e092024f8d26fe9272adec8e1a5f115735aa/lib/connection_config.js#L120

**Comment by wellwelwel** [comment]
> I assumed they should be since they are defined in ConnectionConfig?

@luisasalas, yes, if we consider the typings consistency with the source code, the ideal would be to move them from `PoolOptions` to `ConnectionOptions`.

---

Ref:
https://github.com/sidorares/node-mysql2/blob/8831e092024f8d26fe9272adec8e1a5f115735aa/lib/connection_config.js#L29-L35

**Comment by luisasalas** [comment]
Gotcha, I see it's default to true. However, when using this library in typescript, if we wanted to pass in false, would it be ignored since it's not defined in the typings?

**Comment by wellwelwel** [comment]
> Gotcha, I see it's default to true. However, when using this library in typescript, if we wanted to pass in false, would it be ignored since it's not defined in the typings?

It should work as usual ✅

When using **TypeScript**, it will compile the code to **JavaScript** on build, then the `/lib/connection_config.js` is what will be considered instead of the `Connection.d.ts`.

Also, the declaration types from **MySQL2** are separated from source code (neither depends on the other to work individually). Then these issues are really important to keep the typings consistent with the source code 🧑🏻‍🔧

**Comment by wellwelwel** [comment]
Fixed in `v3.6.4` 🙋🏻‍♂️

Now it will work for both `createConnection`, `createPool` and `createPoolCluster`:

```ts
import mysql, { PoolOptions, ConnectionOptions } from 'mysql2';

const poolOptions: PoolOptions = {
  enableKeepAlive: true,
  keepAliveInitialDelay: 0,
}

const connectionOptions: ConnectionOptions = {
  enableKeepAlive: true,
  keepAliveInitialDelay: 0,
}

mysql.createConnection(connectionOptions);
mysql.createPool(poolOptions);
mysql.createPoolCluster().add(poolOptions);
```

- The same for `mysql2/promise`
